### PR TITLE
Bookmarked history entries now show context menu in about:history

### PR DESCRIPTION
**Pull request by bsclifton**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Fixes https://github.com/brave/browser-laptop/issues/3330

### CODE REVIEW

File: js/state/siteUtil.js [hunk 1]
```diff
@@ -155,12 +155,15 @@ module.exports.removeSite = function (sites, siteDetail, tag) {
...
   if (tags.size === 0 && !tag) {
-    sites = sites.splice(index, 1)
-    return sites
-  }
+    // If called without tags and entry has no tags, remove the entry
+    return sites.splice(index, 1)
+  } else if (tags.size > 0 && !tag) {
+    // If called without tags BUT entry has tags, null out lastAccessedTime.
+    // This is a bookmark entry that we want to clear history for (but NOT delete/untag bookmark)
+    return sites.setIn([index, 'lastAccessedTime'], null)
```
Comment (bbondy) [inline js/state/siteUtil.js]: hah I was going to suggest that as the fix, thanks.

**Comment by bbondy** [comment]
++

**Comment by bridiver** [comment]
I don't think this is a complete fix. We are sending everything in `sites` to `about:history` and that seems incorrect to me. We should only be sending actual history items.

**Comment by bbondy** [comment]
The sites are the last 500 history we keep, when last access time is defined, but there's a follow up for it as a micro optimization to exclude those.

**Pull request merged**